substrate/frame/examples/frame-crate/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use frame::prelude::*;
+
+#[frame::pallet(dev_mode)]
+pub mod pallet {
```

Comment:
We have a lot of docs in the basic example pallet, but none here.  
Makes me wonder if we should maybe `use frame::*` in the basic pallet and remove this one.

Reply:
Would probably make a tracking issue to tackle it later. 

As noted elsewhere in this PR, I want to move all templates into once place, and ASAP all of them using `frame`.

For now, the main purpose of this is to make sure a frame-based crate can compile.